scripts/twister

```diff
@@ -1152,22 +1152,22 @@ def main():
             print(f"Tests found: {count}")
         return
 
-    if VERBOSE > 1 and tplan.discards:
+    if VERBOSE > 1:
         # if we are using command line platform filter, no need to list every
         # other platform as excluded, we know that already.
         # Show only the discards that apply to the selected platforms on the
         # command line
 
-        for i, reason in tplan.discards.items():
-            if options.platform and i.platform.name not in options.platform:
+        for i in tplan.instances.values():
+            if i.status == "filtered" and options.platform and i.platform.name not in options.platform:
```

Comment:
those two conditions couldn't be combined into one, otherwise, it will print all instances as skipped as below:
```
DEBUG   - qemu_x86                  tests/kernel/threads/thread_apis/kernel.threads.apis SKIPPED: Unknown
DEBUG   - qemu_x86                  tests/kernel/threads/thread_apis/kernel.threads.apis.pinonly SKIPPED: In test case platform exclude
DEBUG   - qemu_x86                  tests/kernel/mutex/mutex_api/kernel.mutex          SKIPPED: Unknown
DEBUG   - qemu_x86                  tests/kernel/mutex/mutex_error_case/kernel.mutex_error_case SKIPPED: Unknown
DEBUG   - qemu_x86                  tests/kernel/mutex/sys_mutex/system.mutex          SKIPPED: Unknown
DEBUG   - qemu_x86                  tests/kernel/mutex/sys_mutex/system.mutex.nouser   SKIPPED: Unknown
```

we should check "filtered" status firstly, then check whether it belongs to our expected platforms.

Reply:
Yes, the test cases of the unfiltered instances will also be printed. This is not right.